pkg/controllers/subnet/namespace_handler.go

```diff
@@ -0,0 +1,69 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package subnet
+
+import (
+	"reflect"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/util"
+)
+
+// SubnetSet controller should watch event of namespace, when there are some updates of namespace labels,
+// controller should build tags and update VpcSubnet according to new labels.
+
+type EnqueueRequestForNamespace struct {
+	Client client.Client
+}
+
+func (e *EnqueueRequestForNamespace) Create(_ event.CreateEvent, _ workqueue.RateLimitingInterface) {
+	log.V(1).Info("namespace create event, do nothing")
+}
+
+func (e *EnqueueRequestForNamespace) Delete(_ event.DeleteEvent, _ workqueue.RateLimitingInterface) {
+	log.V(1).Info("namespace delete event, do nothing")
+}
+
+func (e *EnqueueRequestForNamespace) Generic(_ event.GenericEvent, _ workqueue.RateLimitingInterface) {
+	log.V(1).Info("namespace generic event, do nothing")
+}
+
+func (e *EnqueueRequestForNamespace) Update(updateEvent event.UpdateEvent, l workqueue.RateLimitingInterface) {
+	obj := updateEvent.ObjectNew.(*v1.Namespace)
+	if isInSysNs, err := util.IsSystemNamespace(nil, "", obj); err != nil {
+		log.Error(err, "failed to fetch namespace", "namespace", obj.Name)
+		return
+	} else if isInSysNs {
+		log.V(2).Info("namespace is in system namespace, ignore it", "namespace", obj.Name)
+		return
+	}
+	err := reconcileSubnet(e.Client, obj.Name, l)
```

Comment:
Why not put reconcileSubnet in this file and class?

Reply:
OK